packages/db/src/translate/executions.ts

```diff
@@ -0,0 +1,88 @@
+import Knex from 'knex';
+
+import { RecordDoesNotExist } from '@cumulus/errors';
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import Logger from '@cumulus/logger';
+import { PostgresExecution } from '../types/execution';
+import { ExecutionPgModel } from '../models/execution';
+import { CollectionPgModel } from '../models/collection';
+import { AsyncOperationPgModel } from '../models/async_operation';
+
+/**
+ * Translate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} knex
+ *   Knex client
+ * @returns {PostgresExecutionRecord} - converted Execution
+ */
+export const translateApiExecutionToPostgresExecution = async (
+  dynamoRecord: ExecutionRecord,
+  knex: Knex
+): Promise<PostgresExecution> => {
+  const execution = new ExecutionPgModel();
+  const collection = new CollectionPgModel();
+  const asyncOperation = new AsyncOperationPgModel();
+  const logger = new Logger({ sender: '@cumulus/db/translate/executions' });
+
+  // Map old record to new schema.
+  const translatedRecord: PostgresExecution = {
+    async_operation_cumulus_id: (
+      dynamoRecord.asyncOperationId ? await asyncOperation.getRecordCumulusId(
+        knex,
+        { id: dynamoRecord.asyncOperationId }
+      ) : undefined
+    ),
+    status: dynamoRecord.status,
+    tasks: JSON.stringify(dynamoRecord.tasks),
```

Comment:
I'm fairly sure that the `JSON.stringify` is unnecessary when saving objects to the PG database. For some reason it is necessary if you're ever saving an array of values, but that isn't the case for any of these fields

Reply:
Yeah, I believe that's totally right. You should be able to store a JS object in PG. I think it might convert the JS object to a JSON object down the line but either way.

The trick here is our types. I have `JSON.stringify` because some of these fields like `dynamoRecord.tasks` are of type `unknown` on the Dynamo model. I can't assign a field of type `unknown` to a field of type `Object | undefined` as is the case with `tasks` on the PG model, for example. I'm stringifying here as a way of setting the type.

I could maybe convert `dynamoRecord.tasks` (and the rest) to JS objects instead but I think that might add a step and yield the same result as JSON.stringify?

I could also think about changing the type assignment on the Dynamo model. 

Thoughts?

Reply:
So JSON.stringify on `unknown` makes TS recognize it as an `Object`? That is interesting. Regardless, this is a nit, so if it's working, then I just wouldn't bother changing it

Reply:
So circling back to this 😄 

I've updated all of the Execution's usage of `Object` to `object` because `object` will actually give us the validation we want. `Object` will match any primitive object type in addition to object literals. 

The only other place in our PG types where this was an issue was on async operations. That was an easier case because, in the Dynamo to PG translation we weren't trying to go from `unknown` to `object`. In that case it was `string` to `object` so additional updates weren't required. 

Here's the commit if you wouldn't mind a sanity check:

https://github.com/nasa/cumulus/pull/2029/commits/bb393677d17f9a59cb263ede7041b5ed999b5c41

Reply:
Your changes look good to me, though I question why we're using `uknown` instead of `object` on the Dynamo execution record type for those properties

I just tested this and our Dynamo schema validation prevents saving these fields with those properties using anything other than objects:

```
  const foo = {
    arn: 'fake-arn',
    name: 'name',
    error: 1234,
    tasks: 'not-an-object',
    originalPayload: 'not-an-object',
    finalPayload: 'not-an-object',
  };
  await executionModel.create(foo);
```

So I'd be inclined to update the Dynamo record type to replace `unknown -> object` and simplify this even further

Reply:
Yup! We had a separate chat about this. To summarize:

I was thinking we shouldn't change the types in `types/api/executions` because those are in prod and presumably used for validation all over the place. Mark pointed out that the Dynamo schema (/api/models/schemas) is actually more strict with an "object" type already. It should be totally fine to make these match and change the Typescript type for the few "unknown" fields on executions.